### ISSUE TITLE
Herod spawns multiple adds upon being pulled, lever does not close door, adds do not despawn on Herod's death

### ISSUE THREAD

**Issue by Unknown-as-Captain** [elysium-project/server#65]
Remake of https://elysium-project.org/bugtracker/issue/2700 as the Elysium Project bugtracker will soon be deprecated.

## Expected Behavior
Herod has an anti-kiting mechanic that spawns 3 (_three_) Scarlet Myrmidons if you try to kite him out of the room. These adds **only** spawn if Herod moves outside the room, otherwise, they should not spawn.

## Bugged Behavior
Herod currently always spawns 4 (_four_) Scarlet Myrmidons immediately upon being pulled. They always aggro as soon as they spawn, no way to avoid them.
Examining the script for this bossfight shows that [you should be able to prevent the adds from aggroing by having the door closed when pulling Herod](https://github.com/elysium-project/server/blob/19e7622bd3f3d2c0b1bfe65892f25d035b9e222b/src/scripts/eastern_kingdoms/tirisfal_glades/scarlet_monastery/boss_herod.cpp#L111), however, currently the lever to close the door is not working either, so you can't close the door to prevent these adds from aggroing at all.
Furthermore, [the script also shows that these adds are supposed to despawn on Herod's death](https://github.com/elysium-project/server/blob/19e7622bd3f3d2c0b1bfe65892f25d035b9e222b/src/scripts/eastern_kingdoms/tirisfal_glades/scarlet_monastery/boss_herod.cpp#L157), this is currently not working on the live server, the adds remain alive. (Unfortunately I don't know if this is blizzlike behavior or not, I was unable to find any videos of the Myrmidons spawning during Herod's fight, the anti-kiting mechanic was extremely obscure during Retail as it was rarely relevant)

So this is a weird issue where the server's script has some weird not-blizzlike mechanic centered around the door, which, unfortunately, is also bugged. Even if it wasn't bugged, there's still a few issues with it: You could simply close the door, pull Herod, then open the door and kite him out of the room with no adds to stop you. Also, since this is a custom mechanic (not blizzlike), many players simply do not expect it to happen, closing the door was never a part of the scripting in Retail, so players never learned to do it.

## Sources
Wowpedia details the anti-kiting mechanic: http://wow.gamepedia.com/Herod#Strategies
>Also, if you try to run outside the room, Herod will sprint, increasing his movement speed by 60% and almost certainly catching up to you; if you don't get affected by this, 3 Scarlet Myrmidons will spawn and block your path.

Bossfight against Herod in 2006 retail: You can see the door is wide open and no mobs spawn https://www.youtube.com/watch?v=s-EJO3C5EiY

Bossfight against Herod in 2017 Elysium Project - Elysium Server: You can see the un-blizzlike behavior with the mobs spawning on pull, despite Herod never leaving the room https://www.youtube.com/watch?v=7llVPhMlCWE

Thank you for looking into this bug, it's rather important as it makes the instance extremely difficult for the intended level range.

**Comment by ekohl** [comment]
Can confirm this is still an issue.

**Comment by sipertruk** [comment]
The lever door is working for me.
About the spawn I haven't found any video / guide produced at vanilla era mentioning this mechanism.
On the contrary on videos people aren't seem to take any precaution indicating it exists: tank rush in and jump, door stays open.
IF it ever existed in vanilla, in any case they didn't spawn at aggro time and rush in, so why not spawn them only if Herod's target exits the room?

**Comment by Unknown-as-Captain** [comment]
Some more info after I finally did Arm again:
The lever seemed to be working this time. No idea what broke it the first time around.
While the door is closed, the adds don't ~automatically~ aggro, however, they instantly LoS aggro THROUGH the closed door! We were able to eventually avoid all but one of the adds by having everyone stand far enough inside the room so as to not LoS them.
The lone add that we pulled did not despawn on Herod's death, the rest did despawn correctly.
